Linux: warning: multi-character character constant

We are getting this warning on linux:

```
[365/366] Linking CXX executable bin/Release/msquiccoretest
In file included from /builds/kasperk81/aports/community/libmsquic/src/msquic-2.5.5/src/test/MsQuicTests.h:16,
                 from /builds/kasperk81/aports/community/libmsquic/src/msquic-2.5.5/src/test/bin/quic_gtest.h:11,
                 from /builds/kasperk81/aports/community/libmsquic/src/msquic-2.5.5/src/test/bin/quic_gtest.cpp:8:
/builds/kasperk81/aports/community/libmsquic/src/msquic-2.5.5/src/inc/msquic.hpp:120:37: warning: multi-character character constant [-Wmultichar]
  120 | template<typename T, uint32_t Tag = 'lPxC', bool Paged = false>
      |                                     ^~~~~~
/builds/kasperk81/aports/community/libmsquic/src/msquic-2.5.5/src/test/bin/quic_gtest.cpp:18:62: note: '#pragma message: Test compiled with preview features enabled'
   18 | #pragma message("Test compiled with preview features enabled")
      |                                                              ^
[366/366] Linking CXX executable bin/Release/msquictest
```

`Tag` is uint32_t and single quoted string should contain single character.

--- a/src/inc/msquic.hpp
+++ b/src/inc/msquic.hpp
@@ -138,6 +138,7 @@ constexpr _Ty&& CxPlatForward(
 }
 
 #ifdef __GNUC__
+#pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wmultichar" // Multi-character constant used intentionally for the tag
 #endif
 template<typename T, uint32_t Tag = 'lPxC', bool Paged = false>
@@ -162,6 +163,9 @@ public:
         }
     }
 };
+#ifdef __GNUC__
+#pragma GCC diagnostic pop
+#endif
 
 #ifdef CXPLAT_HASH_MIN_SIZE
 

